iquip/apps/builder.py

```diff
@@ -574,6 +582,110 @@ def scanArguments(self) -> Dict[str, Any]:
         }
 
 
+class _CenterScan(_BaseScan):
+    """Widget for center scan in _ScanEntry.
+
+    Attributes:
+        centerSpinBox: QDoubleSpinBox for center argument inside state.
+        spanSpinBox: QDoubleSpinBox for span argument inside state.
+        stepSpinBox: QDoubleSpinBox for step argument inside state.
+        randomizeCheckBox: QCheckBox for randomize argument inside state.
+    """
+    def __init__(
+        self,
+        procdesc: Dict[str, Any],
+        state: Dict[str, Any],
+        parent: Optional[QWidget] = None
+        ):
+        """Extended.
+
+        Args:
+            state: Each key and its value are:
+              center: The center point for the CenterScan sequence.
+              span: The length of the RangeScan sequence.
+              step: The size of step between each number at the RangeScan sequence.
+              randomize: The boolean value that decides whether to shuffle the CenterScan sequence.
+        """
+        super().__init__(procdesc, parent=parent)
+        self.centerSpinBox = QDoubleSpinBox(self)
+        self.applyProperties(self.centerSpinBox)
+        self.centerSpinBox.setValue(state["center"] / self.scale)
+        self.spanSpinBox = QDoubleSpinBox(self)
```

Comment:
As I know, the span means the length of sequence, hence it should be an integer.
Why did you make its spin box `QDoubleSpinBox` instead of `QSpinBox`?

Reply:
Since the center scan is non integer, span could be non integer.
The range of center scan is defined as [center - span/2, center + span/2].

Reply:
Oh, then does "Length of scanning sequence" in snu-quiqcl/artiq-proxy#111 mean the difference between the (possible) maximum and mininum, not the number of sequence items?

Reply:
As I understand, yes.
I made simple example with integer value at https://github.com/snu-quiqcl/artiq-proxy/discussions/111.

Reply:
That example looks little vague becuase its step is 1, so one may think that scan means the number of items.
Please modify it more concretely.